Sources/Prometheus/MetricTypes/Gauge.swift

```diff
@@ -68,12 +70,61 @@ public class PromGauge<NumType: Numeric, Labels: MetricLabels>: PromMetric, Prom
         }
     }
     
+    /// Sets the Gauge to the current unixtime in seconds
+    ///
+    /// - Parameters:
+    ///     - labels: Labels to attach to the value
+    ///
+    /// - Returns: The value of the Gauge attached to the provided labels
+    @discardableResult
+    public func setToCurrentTime(_ labels: Labels? = nil) -> NumType {
+        return self.set(.init(Date().timeIntervalSince1970), labels)
+    }
+    
+    /// Tracks in progress blocks of code or functions.
+    ///
+    ///     func someFunc() -> String { return "ABC" }
+    ///     let newFunc = myGauge.trackInprogress(someFunc)
+    ///     newFunc() // returns "ABC" and increments & decrements Gauge
+    ///
+    /// - Parameters:
+    ///     - labels: Labels to attach to the value
+    ///     - body: Function to wrap progress tracker around
+    ///
+    /// - Returns: The same type of function passed in for `body`, but wrapped to track progress.
+    @inlinable
+    public func trackInprogress<T>(_ labels: Labels? = nil, _ body: @escaping () throws -> T) -> (() throws -> T) {
```

Comment:
Hmmm, yeah I see this is from the Python one directly since:

> A way to track in-progress requests in some piece of code/function. This is `track_inprogress` in Python.


Man the naming here is weird though... 😕 Do you think spelling it as `trackInProgress` would be more "swifty"?